src/classifier/Pipe.h

```diff
@@ -252,15 +277,22 @@ class Pipe {
   Options *options_; // Classifier options.
   Dictionary *dictionary_; // Dictionary for the classifier.
   Reader *reader_; // Reader for reading instances from a file.
-  Writer* writer_; // Writer for writing instance to a file.
-  Decoder* decoder_; // Decoder for this classification task.
+  Writer *writer_; // Writer for writing instance to a file.
+  Decoder *decoder_; // Decoder for this classification task.
   Parameters *parameters_; // Parameter vector.
-  vector<Instance*> instances_; // Set of training instances.
+  vector<Instance*> instances_; // Set of instances.
+  vector<Instance*> formatted_instances_;
+  vector<Instance*> output_instances_;
+
 
   // Number of mistakes and number of total parts at test time (used for
   // evaluation purposes).
   int num_mistakes_;
   int num_total_parts_;
+  
+  bool running_multithreaded_;
+  std::mutex evaluation_lock_;
+  vector<std::thread> list_of_threads_;
```

Comment:
Change this variable name to "threads_". It's not a list, but a vector, and just calling it threads is fine.